src/web/test-helpers/generic-builder.ts

```diff
@@ -0,0 +1,36 @@
+import { Course, Instructor, JoinState } from '../types/api-output';
+
+type GenericBuilder<T> = {
+  [K in keyof T]: (value: T[K]) => GenericBuilder<T>;
+} & { build(): T };
+
+export function createBuilder<T extends object>(initialValues: T): GenericBuilder<T> {
```

Comment:
Would it be okay to add a inline doc explanation of what is `createBuilder` for and a simple example usage of instructor/course builder? Thanks!